components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java

```diff
@@ -2886,6 +2887,399 @@ private OMElement queryFirstAccess(String columnFamily)
         }
     }
 
+    /** Given API name and Application, returns throttling request counts over time for a given time span
+     * 
+     * @param apiName Name of the API
+     * @param provider Provider name
+     * @param appName Application name
+     * @param fromDate Start date of the time span
+     * @param toDate End date of time span
+     * @param groupBy Group by parameter. Supported parameters are :day,hour
+     * @return Throttling counts over time
+     * @throws APIMgtUsageQueryServiceClientException
+     */
+    public List<APIThrottlingOverTimeDTO> getThrottleDataOfAPIAndApplication(String apiName, String provider,
+            String appName, String fromDate, String toDate, String groupBy)
+            throws APIMgtUsageQueryServiceClientException {
+
+        if (dataSource == null) {
+            throw new APIMgtUsageQueryServiceClientException("BAM data source hasn't been initialized. Ensure " +
+                    "that the data source is properly configured in the APIUsageTracker configuration.");
+        }
```

Comment:
The SQL at line 2843 could be a potential SQL injection. The reason is, since this uses the Statement class SQL injection is easily possible, because it compiles the query dynamically by  concatenating parameters. These parameters could be malicious parameters. The way to prevent this is to use prepared statements instead of the Statement class because prepared statements precompile the SQL query and binds the variables to the query correctly.